### ISSUE TITLE
Error: Can't find child node to replace

### ISSUE THREAD

**Issue by jackkamm** [scrm/scrm#104]
The following command yields an error:
```
$SCRM_PATH 44 1000 -t 10 -r 10 10000 -I 6 10 10 0 0 10 10 -en 0 4 1.0 -en 0 3 1.0 -en 0 5 10.0 -en 0 2 50.0 -en 0 6 20.0 -en 0 1 1.0 -eg 0 1 23.0258509299 -eg 0 2 28.7823136624 -eg 0 6 28.7823136624 -es 0.18 1 0.98 -ej 0.18 7 4 -eI 0.2 0 0 2 2 0 0 0 -ej 0.2 1 2 -ej 0.24 6 2 -en 0.24 2 0.25 -es 0.28 2 0.98 -ej 0.28 8 3 -ej 0.4 2 5 -ej 0.8 4 3 -ej 1.2 5 3 -seed 950314410 781939763 574034435 > foo
Error: Can't find child node to replace
Try 'scrm --help' for more information.
```
If I remove the flag `-r 10 10000` (so no recombination), then the command runs without error.

Specifying `-l` yields slightly different but related error messages:
```
Error: Node iterator out of range
```
or
```
Error: Node has no next node
```

**Comment by paulstaab** [comment]
Thanks also for this report, sorry that scrm is giving you so much trouble. The new bug is not related to  the old one, and does also occur with versions before 1.7.0. It is however also related to the -eI option. Right after the first archaic sequence is added, and scrm tries to prune a node that is not part of the tree. That is very strange, not only because the node is not in the tree, but also because there should be no pruning when building the initial tree at the left end of the sequence. The mysterious node was used in previous repetitions, and is probably a leftover that doesn't get removed correctly.

I'll try to narrow it down, and will keep you updated.

**Comment by paulstaab** [comment]
Was cache invalidation again... Nodes could end up in the contemparies of the next locus when using "-eI". Should now be prevented.

**Comment by jackkamm** [comment]
I checked out the new branch and it appears to work now. Thanks very much for the fast response, and for all the excellent work on the package!

I know all too well how bugs can lurk in lesser-used features such as "-eI", but IMO it is a really important feature that is missing from ms, and it is great that you have implemented it. I will let you know if I run across more issues.

**Comment by paulstaab** [comment]
I just released the fixed version 1.7.2.
